src/robotFactory.js

```diff
@@ -1,10 +1,99 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords = {}, chipVersion) {
+    if (coords.x === undefined) {
+      coords.x = 0;
+    }
```

Comment:
```suggestion
    this.coords = {
      x: coords.x || 0,
      ......
    };
```
You can add all coords at once 
![image](https://github.com/mate-academy/js_robot_factory/assets/96409563/d1618873-12c7-46c2-8d72-89f3cc41e04f)